api/kyverno/v1/common_types.go

```diff
@@ -422,6 +426,80 @@ type PodSecurityStandard struct {
 	Images []string `json:"images,omitempty" yaml:"images,omitempty"`
 }
 
+// CEL defines the CEL expression which is used to apply the validation.
+type CEL struct {
+	// Expressions contain CEL expressions which is used to apply the validation.
+	Expressions []CELExpression `json:"expressions,omitempty" yaml:"expressions,omitempty"`
+
+	// ParamKind specifies the kind of resources used to parameterize this policy.
+	// If absent, there are no parameters for this policy and the param CEL variable will not be provided to validation expressions.
+	// If ParamKind refers to a non-existent kind, this policy definition is mis-configured and the FailurePolicy is applied.
+	// +optional
+	ParamKind *ParamKind `json:"paramKind,omitempty" yaml:"paramKind,omitempty"`
```

Comment:
Can we directly use the K8s API `ParamKind` here?

Reply:
the K8s API `ParamKind` contains the apiVersion and kind only. There's `ParamRef` that contains the name and the namespace.

we can either include everything in paramKind as it exists now or we can make use of both the K8s APIs `ParamKind` and `ParamRef`.
It'll be as follows:
```
validate:
   cel:
       paramKind: 
            apiVersion: rules.example.com/v1
            kind: ReplicaLimit
       paramRef:
            name: "replica-limit-test.example.com"
            namespace: "default"
```

Reply:
I see. I think it will be best to stay as close as possible to the ValidatingAdmissionPolicy, and reuse, upstream definitions. WDYT?

Reply:
I agree.